### PR TITLE
Add Arrow format doc

### PR DESCRIPTION
**Pull request by cih9088**
### Summary
Add a doc for arrow format with binary depicted.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [x] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: docs/source/docs/explanation/formats/arrow.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
+# Arrow
+
+## Format specification
+
+[Apache Arrow](https://arrow.apache.org/docs/format/Columnar.html) is a in-memroy columnar table format specification with multiple langauge support.
```
Comment (vinnamkim) [inline docs/source/docs/explanation/formats/arrow.md]:
```suggestion
[Apache Arrow](https://arrow.apache.org/docs/format/Columnar.html) is a in-memory columnar table format specification with multiple language support.
```

File: docs/source/docs/explanation/formats/arrow.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+### media.bytes (`binary`)
+The binary data of the media. It could be `None` if one chooses not to save media when export.
+
+### media.attributes (`binary`)
+The attribute of the entity. The contents of it depends on `media.type`.
+The byte order is little-endian.
+
+**Image**
+
+![image attributes](../../../../images/arrow/image_attributes.png)
```
Comment (vinnamkim) [inline docs/source/docs/explanation/formats/arrow.md]: Why don't we move this image directory to `docs/source/docs/explanation/formats`?

File: docs/source/docs/explanation/formats/arrow.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+    ├── <subset_name_1>-1-of-2.arrow
+    ├── <subset_name_2>-0-of-1.arrow
+    └── ...
+```
+
+If your dataset is not following the above directory structure,
+it cannot detect and import your dataset as the Arrow format properly.
+
+To make sure that the selected dataset has been added to the project, you can
+run `datum project info`, which will display the project information.
```
Comment (vinnamkim) [inline docs/source/docs/explanation/formats/arrow.md]:
```suggestion
run `datum project pinfo`, which will display the project information.
```

File: docs/source/docs/explanation/formats/arrow.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+There are several ways to convert a Datumaro dataset to other dataset formats
+using CLI:
+
+- Export a dataset from Datumaro format to VOC format:
+
+```console
+datum create
+datum import -f arrow <path/to/dataset>
+datum export -f voc -o <output/dir>
```
Comment (vinnamkim) [inline docs/source/docs/explanation/formats/arrow.md]:
```suggestion
datum project create
datum project import -f arrow <path/to/dataset>
datum project export -f voc -o <output/dir>
```

File: docs/source/docs/explanation/formats/arrow.md [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+```
+
+## Export to Arrow
+
+There are several ways to convert a dataset to Arrow format:
+
+- Export a dataset from an existing project to Arrow format:
+```console
+# export dataset into Arrow format from existing project
+datum export -p <path/to/project> -f arrow -o <output/dir> \
```
Comment (vinnamkim) [inline docs/source/docs/explanation/formats/arrow.md]:
```suggestion
datum project export -p <path/to/project> -f arrow -o <output/dir> \
```

File: docs/source/docs/explanation/formats/arrow.md (hunk 2 above)
Comment (cih9088) [reply docs/source/docs/explanation/formats/arrow.md]: I assumed that images are in top level as other images in ‘image’ directory. Do you want me to move them?

File: docs/source/docs/explanation/formats/arrow.md (hunk 4 above)
Comment (cih9088) [reply docs/source/docs/explanation/formats/arrow.md]: I copied them from datumaro_binary.md. Does it also need to be changed? I will, if so.

Comment (vinnamkim) [reply docs/source/docs/explanation/formats/arrow.md]: We need to do it for all formats in a separate PR.

File: docs/source/docs/explanation/formats/arrow.md (hunk 2 above)
Comment (vinnamkim) [reply docs/source/docs/explanation/formats/arrow.md]: I think that's cleaner.

Comment (cih9088) [reply docs/source/docs/explanation/formats/arrow.md]: Moved in https://github.com/openvinotoolkit/datumaro/pull/948/commits/187f2ad1c1f3b46346ba1939d8590911a1bd08d7

**Review by vinnamkim — APPROVED** [review]
LGTM.

**Review by sooahleex — APPROVED** [review]

**Review by bonhunko — APPROVED** [review]

**Pull request merged**